dispatch/dispatch.go

```diff
@@ -133,7 +138,17 @@ func (d *Dispatcher) Groups(matchers []*labels.Matcher) AlertOverview {
 				if !a.EndsAt.IsZero() && a.EndsAt.Before(now) {
 					continue
 				}
+
 				status := d.marker.Status(a.Fingerprint())
+
+				if !showSilenced && len(status.SilencedBy) != 0 {
+					continue
+				}
+
+				a.Annotations = a.Annotations.Merge(
+					model.LabelSet{"receiver": model.LabelValue(route.RouteOpts.Receiver)},
```

Comment:
The snippet you posted is the reverse of what I was thinking. I was thinking you have an index/map from receiver to the possible leaf routes that have that would send notifications to that receiver.

Maybe we're making this too complicated. I'm thinking the receivers could just be a short-hand to add values to the search query. As in you click on the receiver in the UI and it just adds the possible label queries that can lead to that receiver to the search, because in the end a receiver is just an abstraction/union of routes. I feel that would be the ideal UX, but I'm unsure about the complexity of this in terms of changes in the backend.